async/as/ThreadBase.h

```diff
@@ -44,13 +44,34 @@
 #include "utils/logger.h"
 
 #include "Base.h"
+#include "Pin.h"
 
 namespace async
 {
 
 namespace as
 {
 
+#ifndef __APPLE__
```

Comment:
Here, the `#ifndef` may look a bit confusing. Maybe `#ifdef` it instead (and swap the execution branches)? Maybe make the other one check for Linux? Or is the Apple-path POSIX-compliant?

Reply:
You cannot check for Linux :(

Reply:
Also, ifndef is by design: The most relevant code path comes first. (And any reasonable IDE will show only that codepath anyway)

Reply:
https://gitlab.kitware.com/cmake/community/-/wikis/doc/tutorials/How-To-Write-Platform-Checks#platform-variables
This is the only way to check for Linux (there aren't any compiler standard defs). And the CMake platform is not documented. The proper way would be doing feature checking which is a ton of effort and doesn't really have a strong advantage